faasmcli/faasmcli/tasks/codegen.py

```diff
@@ -27,31 +28,35 @@ def codegen(ctx, user, function, wamr=False):
         }
     )
 
+    if (not wamr) and sgx:
+        print("Can't run SGX codegen with WAVM. Add --wamr flag.")
+        exit(1)
+
     binary = find_codegen_func()
     run(
-        "{} {} {}".format(binary, user, function),
+        "{} {} {} {}".format(binary, user, function, "--sgx" if sgx else ""),
         shell=True,
         env=env,
         check=True,
     )
 
 
 @task
-def user(ctx, user, wamr=False):
```

Comment:
These optional parameters were really never used, and they would not work neither.